packages/iTwinUI-react/src/core/Header/HeaderSplitButton.tsx

```diff
@@ -0,0 +1,96 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import cx from 'classnames';
+import React from 'react';
+import { DropdownMenu } from '../DropdownMenu';
+import SvgCaretDownSmall from '@itwin/itwinui-icons-react/cjs/icons/CaretDownSmall';
+import SvgCaretUpSmall from '@itwin/itwinui-icons-react/cjs/icons/CaretUpSmall';
+
+import { PolymorphicForwardRefComponent, useTheme } from '../utils';
+import { SplitButtonProps } from '../Buttons';
+
+export type HeaderSplitButtonProps = SplitButtonProps;
+
+type HeaderSplitButtonComponent = PolymorphicForwardRefComponent<
+  'button',
+  HeaderSplitButtonProps
+>;
+
+export const HeaderSplitButton: HeaderSplitButtonComponent = React.forwardRef(
+  (props, forwardedRef) => {
+    const {
+      onClick,
+      menuItems,
+      className,
+      menuPlacement = 'bottom-end',
+      size,
+      children,
+      style,
+      title,
+      type = 'button',
+      startIcon,
+      ...rest
```

Comment:
In here we also have `endIcon` and `styleType`, which are now not working, but if you pass those through on `button` they will be set in HTML and will be as invalid attributes :/
So I think we need to destruct them out of props and not use?
Oh, there is also polymorphism, which is now also not working anymore :/

Reply:
`styleType` has been removed as a prop since `HeaderButton` only has the `styleType: borderless`. The previous `HeaderButton` did not have a `styleType` prop, so this should not cause any breaking changes.

`endIcon` support has now been added. It is passed into `HeaderBasicButton` and now `HeaderDropdownButton` and `HeaderSplitButton` both use `HeaderBasicButton` instead of the `<button>` element

Reply:
amm... so if you plan to reuse `HeaderBasicButton`, you can get rid of destructing explicitly here `styleType`, `size` and leave those in `rest`, so they are passed to `HeaderBasicButton` anyway..
But whats important, is to remove these props from `rest` in `HeaderBasicButton`. So you need to desctruct them and not use, otherwise, you get button rendered in DOM like that
![image](https://user-images.githubusercontent.com/81580355/183952747-6daad313-2552-42fb-8517-612981158d02.png)

So need to review all props which do not work and take care of those.

Also, I do not see here `endIcon` being passed to `HeaderBasicButton` as you mentioned in your comment.

Reply:
`styleType` and `size` are destructed in `HeaderBasicButton` and not used. I decided to just call the variables inside of `HeaderBasicButton` instead of adding eslint ignore to the file.

[Here](https://github.com/iTwin/iTwinUI-react/pull/755/files#diff-d07cba2fdef199f25e834f5c92fe6046a74a905ed2e0e07148b7bbd20b52f6f5R26) is where I destructed `endIcon` inside of `HeaderBasicButton`.

Reply:
What about `disabled` ?

Reply:
@adamhock how is it going with `disabled` support? are you facing any issues?

Reply:
`disabled` support does exist for `HeaderBasicButton`. I added `disabled` support for the split menu of `HeaderSplitButton` by destructing it in `HeaderSplitButton` and passing the prop to the `HeaderBasicButton` and `<button>` elements.

The `disabled` state for `HeaderButton` is not the same as it was because the CSS changes for `Header` only apply `cursor: not-allowed` to `iui-header-breadcrumb-button`. The text and background colors are not changed, and we should fix this in CSS